src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -1307,6 +1316,13 @@ private void CloseResponseBody()
                 {
                     Cancel();
                 }
+                else if (_sendRstOnResponseClose)
```

Comment:
This is called with response `Dispose`. Which works great for cancelling partially consumed responses. But for fully consumed response - if the user doesn't Dispose the response - we would be waiting for GC to collect the stream and send the RST. Shouldn't we rather send it immediately after we reach the end of response stream?

Note that I'm not saying this is wrong, just thinking of possible scenarios that could happen.

Reply:
Hmm. You are right, delaying sending seems as an issue. Do you know how and where we can react to reaching the end of response stream?

Reply:
I'd say in this else block:
https://github.com/dotnet/runtime/blob/49ea27e3344f530db56ca5cba5d1c61af5f6f91f/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs#L1114-L1118